addons/product/models/product.py

```diff
@@ -357,7 +357,8 @@ def unlink(self):
             if not product.exists():
                 continue
             # Check if the product is last product of this template...
-            other_products = self.search([('product_tmpl_id', '=', product.product_tmpl_id.id), ('id', '!=', product.id)])
+            other_products = self.search([('product_tmpl_id', '=', product.product_tmpl_id.id), ('id', '!=', product.id),\
```

Comment:
Hi @Alexcn there are a better way to search on all records:
`other_products = self.with_context(active_test=False).search([('product_tmpl_id', '=', product.product_tmpl_id.id), ('id', '!=', product.id)])`

Reply:
I think it must be:
`other_products = self.with_context(active_test=False).search([('product_tmpl_id', '=', product.product_tmpl_id.id), ('id', '!=', self.ids)])`

Because if you delete all variants of a template, other_products is allways returning a record and the variant is not deleted

Reply:
I think may use a better way:

```
other_products = self.env['product.product'].with_context(active_test=False).\
                                    search([('product_tmpl_id', '=', product.product_tmpl_id.id), ('id', 'not in', self.ids)])
```

Beacuse `('id', '!=', self.ids)` could trigger a warning:

![image](https://user-images.githubusercontent.com/16533962/50961639-2e74c300-1503-11e9-9334-f0878a76750e.png)

My new PR: [https://github.com/odoo/odoo/pull/30094](https://github.com/odoo/odoo/pull/30094)